doc/qos/SONiC_ROCEv2_support.md

```diff
@@ -0,0 +1,574 @@
+
+#### 1.1.1.5 Get operation on QoS buffer and it's mapping
+This displays the output of the buffer pool/profile and it's mapping with priority group and queue.
+
+#### 1.1.1.6 Allow user defined lossless profile based on cable length and speed
+This requirement is to add a field in the PORT table to avoid automatic creation of buffer profile
+for lossless profile based on the cable length and speed so that user can create the profile based on the particular use-case.
+
+### 1.1.2 Backend mechanisms to support configuration and get
+ QoS buffer init populates/clears the following tables in the Redis ConfigDB by default, the size & xoff values vary based on HW capabilities.
+
+```
+"BUFFER_POOL": {
+ "egress_lossless_pool": {
+     "mode": "static",
+     "size": "32575488",
+     "type": "egress"
+ },
+ "ingress_lossless_pool": {
+     "mode": "dynamic",
+     "size": "26284032",
+     "type": "ingress",
+     "xoff": "6291456"
+ }
+},
+"BUFFER_PROFILE": {
+ "egress_lossless_profile": {
+     "mode": "static",
+     "pool": "[BUFFER_POOL|egress_lossless_pool]",
+     "size": "0",
+     "static_th": "32575488"
+ },
+ "egress_lossy_profile": {
+     "dynamic_th": "3",
+     "mode": "dynamic",
+     "pool": "[BUFFER_POOL|egress_lossless_pool]",
+     "size": "0"
+ },
+ "ingress_lossy_profile": {
+     "dynamic_th": "3",
+     "pool": "[BUFFER_POOL|ingress_lossless_pool]",
+     "size": "0"
+ },
+ "pg_lossless_100000_40m_profile": {
+     "dynamic_th": "-3",
+     "pool": "[BUFFER_POOL|ingress_lossless_pool]",
+     "size": "1248",
+     "xoff": "177632",
+     "xon": "2288",
+     "xon_offset": "2288"
+ },
+ "pg_lossless_10000_40m_profile": {
+     "dynamic_th": "-3",
+     "pool": "[BUFFER_POOL|ingress_lossless_pool]",
+     "size": "1248",
+     "xoff": "37024",
+     "xon": "2288",
+     "xon_offset": "2288"
+ }
+},
+"BUFFER_PG": {
+       "Ethernet0|0": {
+           "profile": "[BUFFER_PROFILE|ingress_lossy_profile]"
+       },
+       "Ethernet0|3-4": {
+           "profile": "[BUFFER_PROFILE|pg_lossless_100000_40m_profile]"
+       },
+       ........
+       .......
+}
+
+"BUFFER_QUEUE": {
+  "Ethernet0|0-2": {
+          "profile": "[BUFFER_PROFILE|egress_lossy_profile]"
+   },
+  "Ethernet0|3-4": {
+          "profile": "[BUFFER_PROFILE|egress_lossless_profile]"
+   },
+  "Ethernet0|5-6": {
+          "profile": "[BUFFER_PROFILE|egress_lossy_profile]"
+  },
+  .......
+  .......
+}                                                                                                                                                                        48        563,7         19%
+
+```
+The above tables are populated by default already in the SONiC system upon executing
+the 'config qos buffer reload' Click command, we are providing the following configuration options to fine tune the values
+and the get support will be provided to dump the buffer information.
+
+### 1.1.3 Functional Requirements

```

Comment:
Does configuration of buffer pool and buffer profile is enough for validation or should we also test some functional behavior of those commands as part of the testing? Please confirm if only configuration part is enough for test plan preperation.

Reply:
We should verify the ROCEv2 traffic in the system test but functional test can be limited to test the PFC behaviour from IXIA.